### PR TITLE
docs: fit `factorial.cpp` to contribution guidelines

### PR DESCRIPTION
**Pull request by realstealthninja**
#### Description of Change

fixed documentation of factorial.cpp
added much needed tests
added an exception so that numbers larger than an unsigned 64 bit integer is not used as an argument

#### Checklist

- [x] Added description of change
- [ ] ~Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)~
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: part of #2456

### CODE REVIEW

File: math/factorial.cpp [hunk 1]
```diff
@@ -1,20 +1,59 @@
...
+
+/**
+ * @namespace
+ * @brief Mathematical algorithms
+ */
+namespace math {
+
+/**
+ * @brief function to find factorial of given number
+ * @warning Maximum value for the parameter is 20 as 21!
```
Comment (Panquesito7) [inline math/factorial.cpp:25]: Missing `n` parameter description.

File: math/factorial.cpp [hunk 2]
```diff
@@ -1,20 +1,59 @@
 /**
  * @file
- * @brief C++ program to find factorial of given number
+ * @brief Find the [factorial](https://en.wikipedia.org/wiki/Factorial) of a given number
+ * @description calculates factorial via recursion
```
Comment (Panquesito7) [inline math/factorial.cpp:4]: AFAIK, `@description` is invalid. `@details` should be used.
```suggestion
 * @details Calculate factorial via recursion
```

File: math/factorial.cpp [hunk 3]
```diff
@@ -1,20 +1,59 @@
...
+ * \f$5! = 5\times4! = 5\times4\times3\times2\times1 = 120\f$
+ *
+ * @author [Akshay Gupta](https://github.com/Akshay1910)
  */
-#include <iostream>
 
-/** function to find factorial of given number */
-unsigned int factorial(unsigned int n) {
-    if (n == 0)
+#include <cstdint>   /// for uint64_t
```
Comment (Panquesito7) [inline math/factorial.cpp:13]: AFAIK, `uint_*` types are already included without the need of any external libraries.

Comment (realstealthninja) [reply math/factorial.cpp]: ah makes sense my ide keeps freaking out when i dont include it

File: math/factorial.cpp (hunk 1 above)
Comment (realstealthninja) [reply math/factorial.cpp:25]: fixed it !

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀
Will be merging in a few days if there are no objections. In case I forget, feel free to ping me. 🙂

**Comment by realstealthninja** [comment]
@Panquesito7

**Pull request merged**